File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    list_temp = [0] * 46
    list_temp[1] = 1
    list_temp[2] = 2
    def climbStairs(self, n: int) -> int:
        if self.list_temp[n]:
            return self.list_temp[n]
        else:
            int_temp = self.climbStairs(n - 1) + self.climbStairs(n - 2)
            self.list_temp[n] = int_temp
            return int_temp
        

File: 0070-climbing-stairs/0070-climbing-stairs.dart
class Solution {
  int climbStairs(int n) {
    if (n == 1) {return 1;}
    if (n == 2) {return 2;}
    List<int> dp = List.filled(46, 0);
    dp[1] = 1;
    dp[2] = 2;
    for (var i = 3; i <= n; i ++) {
        dp[i] = dp[i-1] + dp[i-2];
    }
    int answer = dp[n];
    return answer;
  }
}